packages/switch/src/mwc-switch-base.ts

```diff
@@ -63,29 +70,95 @@ export class SwitchBase extends FormElement {
     };
   }
 
-  get ripple() {
-    return this.rippleNode.ripple;
+  protected rippleHandlers: RippleHandlers = new RippleHandlers(() => {
+    this.shouldRenderRipple = true;
+    return this.ripple;
+  });
+
+  protected renderRipple() {
+    if (this.shouldRenderRipple) {
+      return html`<mwc-ripple .accent="${this.checked}" .disabled="${
```

Comment:
Mind manually formatting this line so the formatter doesn't mangle the bindings?